app/src/test/kotlin/compiler/lexer/regex/RegexFactoryTest.kt

```diff
@@ -0,0 +1,215 @@
+package compiler.lexer.regex
+
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotEquals
+
+private val ATOMIC_AB = RegexFactory.createAtomic(setOf('a', 'b'))
+private val ATOMIC_BC = RegexFactory.createAtomic(setOf('c', 'b'))
+private val ATOMIC_ABC = RegexFactory.createAtomic(setOf('c', 'a', 'b'))
+private val CONCAT_AB = RegexFactory.createConcat(RegexFactory.createAtomic(setOf('a')), RegexFactory.createAtomic(setOf('b')))
+private val CONCAT_AC = RegexFactory.createConcat(RegexFactory.createAtomic(setOf('a')), RegexFactory.createAtomic(setOf('c')))
+private val CONCAT_BC = RegexFactory.createConcat(RegexFactory.createAtomic(setOf('b')), RegexFactory.createAtomic(setOf('c')))
+private val CONCAT_CD = RegexFactory.createConcat(RegexFactory.createAtomic(setOf('c')), RegexFactory.createAtomic(setOf('d')))
+private val EMPTY = RegexFactory.createEmpty()
+private val EPSILON = RegexFactory.createEpsilon()
+
+class RegexFactoryTest {
+    private fun <T> assertAllEqual(elements: List<T>) {
+        if (elements.size == 0) return
+        elements.forEach { assertEquals(it, elements[0]) }
+    }
+
+    @Test fun `concat preserves order`() {
```

Comment:
In the other PRs we start naming test with a word "test', so to be consistent, we might do it there also.